geoai/universat.py

```
@@ -0,0 +1,230 @@
+        raise ValueError(f"Expected 2D or 3D tensor, got {tokens.shape}")
+    return tokens.mean(dim=1 if tokens.ndim == 3 else 0)
+
+
+def get_pca_rgb(tokens: torch.Tensor) -> np.ndarray:
+    t = (
+        tokens.detach().cpu().numpy()
+        if isinstance(tokens, torch.Tensor)
+        else np.asarray(tokens)
+    )
+
+    def _proj(x, g):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_proj`

(ANN202)